core/geotime/src/test/java/org/locationtech/geowave/core/geotime/adapter/annotation/SpatialTemporalAnnotationsTest.java

```diff
@@ -0,0 +1,138 @@
+import org.locationtech.jts.geom.Geometry;
+import org.opengis.referencing.FactoryException;
+import org.opengis.referencing.NoSuchAuthorityCodeException;
+import org.opengis.referencing.crs.CoordinateReferenceSystem;
+
+public class SpatialTemporalAnnotationsTest {
+
+  private static final String TEST_CRS_CODE = "EPSG:3857";
+
+  @Test
+  public void testSpatialTemporalAnnotations()
+      throws NoSuchAuthorityCodeException, FactoryException {
+    BasicDataTypeAdapter<TestType> adapter =
+        BasicDataTypeAdapter.newAdapter("myType", TestType.class, "name");
+
+    final CoordinateReferenceSystem testCRS = CRS.decode(TEST_CRS_CODE);
+    assertEquals("myType", adapter.getTypeName());
+    assertEquals(TestType.class, adapter.getDataClass());
+    assertEquals(3, adapter.getFieldDescriptors().length);
+    assertNotNull(adapter.getFieldDescriptor("name"));
+    assertTrue(String.class.isAssignableFrom(adapter.getFieldDescriptor("name").bindingClass()));
+    SpatialFieldDescriptor<?> geometryDescriptor =
+        (SpatialFieldDescriptor<?>) adapter.getFieldDescriptor("geometry");
+    assertNotNull(geometryDescriptor);
+    assertTrue(Geometry.class.isAssignableFrom(geometryDescriptor.bindingClass()));
+    assertTrue(geometryDescriptor.indexHints().contains(SpatialField.LATITUDE_DIMENSION_HINT));
+    assertTrue(geometryDescriptor.indexHints().contains(SpatialField.LONGITUDE_DIMENSION_HINT));
+    assertEquals(testCRS, geometryDescriptor.crs());
+    assertNotNull(adapter.getFieldDescriptor("date"));
+    assertTrue(Date.class.isAssignableFrom(adapter.getFieldDescriptor("date").bindingClass()));
+    assertTrue(
+        adapter.getFieldDescriptor("date").indexHints().contains(TimeField.TIME_DIMENSION_HINT));
+
+    final byte[] adapterBytes = PersistenceUtils.toBinary(adapter);
+    adapter = (BasicDataTypeAdapter) PersistenceUtils.fromBinary(adapterBytes);
+
+    assertEquals("myType", adapter.getTypeName());
+    assertEquals(TestType.class, adapter.getDataClass());
+    assertEquals(3, adapter.getFieldDescriptors().length);
+    assertNotNull(adapter.getFieldDescriptor("name"));
+    assertTrue(String.class.isAssignableFrom(adapter.getFieldDescriptor("name").bindingClass()));
+    geometryDescriptor = (SpatialFieldDescriptor<?>) adapter.getFieldDescriptor("geometry");
+    assertTrue(Geometry.class.isAssignableFrom(geometryDescriptor.bindingClass()));
+    assertTrue(geometryDescriptor.indexHints().contains(SpatialField.LATITUDE_DIMENSION_HINT));
+    assertTrue(geometryDescriptor.indexHints().contains(SpatialField.LONGITUDE_DIMENSION_HINT));
+    assertEquals(testCRS, geometryDescriptor.crs());
+    assertNotNull(adapter.getFieldDescriptor("date"));
+    assertTrue(Date.class.isAssignableFrom(adapter.getFieldDescriptor("date").bindingClass()));
+    assertTrue(
+        adapter.getFieldDescriptor("date").indexHints().contains(TimeField.TIME_DIMENSION_HINT));
+
+    // final TestType testEntry = new TestType("id1", 2.5, 8, true);
```

Comment:
why the commented out block?

Reply:
Oops, this was an oversight, I will update this test.